### ISSUE TITLE
模型复现的效果怎么样呢？

### ISSUE THREAD

**Issue by aceyw** [cowarder/Poly-YOLO#2]
您好，请问这个模型代码复现的效果怎样呢，和论文里的一样吗

**Comment by Tarrin** [comment]
我跑了一下，loss 一直不收敛，不知道是哪里出问题了

**Comment by dengbuqi** [comment]
我觉得核心的问题所在是这个F.binary_cross_entropy_with_logits()应该加上reduction='none'参数。否则计算的结果会求平均值，输出loss只有一个值，无法和后面的那些object_mask做element-wise的乘法.
F.binary_cross_entropy_with_logits(predictions,targets) -> F.binary_cross_entropy_with_logits(predictions,targets,reduction='none')

**Comment by Tarrin** [comment]
您的邮件已收到。谢谢您的来件！

**Comment by YoungTaoD** [comment]
加上reduction好像也不能收敛呢

**Comment by dengbuqi** [comment]
@dyt0414 我魔改了好多，现在效果也不太好，而且，loss里面的confidence_loss也老是很大，另外，F.binary_cross_entropy_with_logits()里面ground truth的值也不能有负数，不然也有问题。至于你的不收敛的问题，可能还得具体问题具体分析了。

**Comment by UOSFZ** [comment]
> 我跑了一下，loss 一直不收敛，不知道是哪里出问题了

我的也是，而且loss还是负数。。。

**Comment by UOSFZ** [comment]
> 我觉得核心的问题所在是这个F.binary_cross_entropy_with_logits()应该加上reduction='none'参数。否则计算的结果会求平均值，输出loss只有一个值，无法和后面的那些object_mask做element-wise的乘法. F.binary_cross_entropy_with_logits(predictions,targets) -> F.binary_cross_entropy_with_logits(predictions,targets,reduction='none')

请问您能分享一下吗 魔改后的

**Comment by dengbuqi** [comment]
> > 我觉得核心的问题所在是这个F.binary_cross_entropy_with_logits()应该加上reduction='none'参数。否则计算的结果会求平均值，输出loss只有一个值，无法和后面的那些object_mask做element-wise的乘法. F.binary_cross_entropy_with_logits(predictions,targets) -> F.binary_cross_entropy_with_logits(predictions,targets,reduction='none')
> 
> 请问您能分享一下吗 魔改后的

对不起啊，＞人＜。暂时不行，等我写完论文了再分享给大家
